cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ZookeeperCluster.java

```diff
@@ -275,7 +290,25 @@ public Secret generateNodesSecret() {
                 .withResources(resources())
                 .build();
 
+        ResourceRequirements resources = new ResourceRequirementsBuilder()
+                .addToRequests("cpu", new Quantity("100m"))
+                .addToRequests("memory", new Quantity("128Mi"))
+                .addToLimits("cpu", new Quantity("1"))
+                .addToLimits("memory", new Quantity("128Mi"))
+                .build();
+
+        Container stunnelContainer = new ContainerBuilder()
+                .withName(STUNNEL_NAME)
+                .withImage(stunnelImage)
+                .withResources(resources)
```

Comment:
I think we need to clarify how are the resources shared between the sidecar and the main container.

Reply:
What do you mean?

Reply:
IIUC the resources of the containers are added and those are then used as the resources for the pod as a whole.

Reply:
So there are some practical problems with this ... As mentioned by @tombentley ... how are we sure that fixed resources will be enough here? 

With the `kafka-init` we know exactly what it does and how much it needs. We also expect this will be significantly smaller than the resources configured for Kafka, so we can reasonably expect that it will be available. And since its an init container, it actually means that its not running together with the main container.

Its a different story here ... the stunnel requiremenets will probably rise with traffic. They will be probably also higher for 7 node cluster than for 3 node cluster (less connections). So maybe we would need to configure it independently so that the stunnel doesn't become our bottleneck. Second problem is that its running together with the main container. So when the user says that ha wants to run Zoo on 4CPU and 8GB or RAM we actually make it 4+ CPU and 8+ GB of RAM. What if this means that the Pod will not be scheduled because this isn't available anymore? Its a bit intransparent to the user.

That said, I'm not sure because I do not want to overcomplicate things. I'm wondering what is the best way to keep this open for future improvements yet functional right now.